### ISSUE TITLE
[Bug] Missing @babel/runtime dependency

### ISSUE THREAD

**Issue by nemanja-tosic** [visgl/deck.gl#7936]
### Description

While working on a project that uses Yarn PnP i've started getting errors during the build saying `Module not found: Can't resolve '@babel/runtime/helpers/interopRequireDefault'`. Looking into the distributed package, it appears that the babel runtime plugin is used, however the @babel/runtime dependency is missing. Currently, it appears that another package is providing @babel/runtime (added below) but this is potentially problematic because the project is using a dependency it does not declare. For reference, adding the output of `yarn why @babel/runtime`

```
yarn why v1.22.19
[1/4] 🤔  Why do we have the module "@babel/runtime"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
[4/4] 🚡  Calculating file sizes...
=> Found "@babel/runtime@7.22.3"
info Reasons this module exists
   - "@probe.gl#env" depends on it
   - Hoisted from "@probe.gl#env#@babel#runtime"
   - Hoisted from "@probe.gl#log#@babel#runtime"
   - Hoisted from "@probe.gl#stats#@babel#runtime"
   - Hoisted from "deck-graph-layers#@luma.gl#core#@babel#runtime"
   - Hoisted from "@deck.gl#core#@luma.gl#webgl#@babel#runtime"
   - Hoisted from "deck-graph-layers#probe.gl#@babel#runtime"
   - Hoisted from "@deck.gl#core#@math.gl#core#@babel#runtime"
   - Hoisted from "@deck.gl#core#@math.gl#sun#@babel#runtime"
   - Hoisted from "@deck.gl#core#@math.gl#web-mercator#@babel#runtime"
   - Hoisted from "@deck.gl#core#@loaders.gl#core#@babel#runtime"
   - Hoisted from "deck-graph-layers#@luma.gl#core#@luma.gl#engine#@babel#runtime"
   - Hoisted from "deck-graph-layers#@luma.gl#core#@luma.gl#gltools#@babel#runtime"
   - Hoisted from "deck-graph-layers#@luma.gl#core#@luma.gl#shadertools#@babel#runtime"
   - Hoisted from "@deck.gl#core#@loaders.gl#core#@probe.gl#log#@babel#runtime"
   - Hoisted from "@deck.gl#core#@loaders.gl#core#@loaders.gl#loader-utils#@babel#runtime"
   - Hoisted from "@deck.gl#core#@loaders.gl#core#@loaders.gl#worker-utils#@babel#runtime"
   - Hoisted from "@deck.gl#core#@loaders.gl#core#@probe.gl#log#@probe.gl#env#@babel#runtime"
   - Hoisted from "@deck.gl#core#@loaders.gl#core#@loaders.gl#loader-utils#@probe.gl#stats#@babel#runtime"
```

### Flavors

- [ ] Script tag
- [X] React
- [ ] Python/Jupyter notebook
- [ ] MapboxOverlay
- [ ] GoogleMapsOverlay
- [ ] CartoLayer
- [ ] ArcGIS

### Expected Behavior

Every package should declare the dependencies it depends on.

### Steps to Reproduce

Use yarn in PnP mode. If a detailed reproduction is necessary i can provide a repository link.

### Environment

- Framework version: 8.9.18
- Browser: /
- OS: /

### Logs

```
../../../.yarn/__virtual__/@deck.gl-react-virtual-cde72c67e7/3/.yarn/berry/cache/@deck.gl-react-npm-8.9.18-c375305247-9.zip/node_modules/@deck.gl/react/dist/es5/deckgl.js
Module not found: Can't resolve '@babel/runtime/helpers/interopRequireDefault' in '/Users/ntosic/Projects/app/.yarn/__virtual__/@deck.gl-react-virtual-cde72c67e7/3/.yarn/berry/cache/@deck.gl-react-npm-8.9.18-c375305247-9.zip/node_modules/@deck.gl/react/dist/es5'

Import trace for requested module:
../../../.yarn/__virtual__/@deck.gl-react-virtual-cde72c67e7/3/.yarn/berry/cache/@deck.gl-react-npm-8.9.18-c375305247-9.zip/node_modules/@deck.gl/react/dist/es5/deckgl.js
[../../../.yarn/__virtual__/@deck.gl-react-virtual-cde72c67e7/3/.yarn/berry/cache/@deck.gl-react-npm-8.9.18-c375305247-9.zip/node_modules/@deck.gl/react/dist/es5/index.js
../../../.yarn/__virtual__/react-deck-graph-layers-virtual-4dbe4fa23b/3/.yarn/berry/cache/react-deck-graph-layers-npm-0.0.6-37011d2792-9.zip/node_modules/react-deck-graph-layers/dist/index.cjs
```

**Comment by Pessimistress** [comment]
This is fixed in the latest release of probe.gl. You need to upgrade all your `@probe.gl/*` dependencies.

**Comment by nemanja-tosic** [comment]
With Yarn PnP, the error comes from any package that uses @babel/runtime but does not declare it as a dependency. The errors come from @deck.gl, @math.gl, @loaders.gl, at least. In order to use react-deck-graph-layers, i had to add the following to yarnrc.yml:

```
packageExtensions:
  "@math.gl/polygon@*":
    dependencies:
      "@babel/runtime": "*"
  "@loaders.gl/images@*":
    dependencies:
      "@babel/runtime": "*"
  "@deck.gl/core@*":
    dependencies:
      "@babel/runtime": "*"
  "@deck.gl/react@*":
    dependencies:
      "@babel/runtime": "*"
  "@deck.gl/layers@*":
    dependencies:
      "@babel/runtime": "*"
  deck-graph-layers@*:
    dependencies:
      "@babel/runtime": "*"
      "@loaders.gl/core": "*"
  react-deck-graph-layers@*:
    dependencies:
      "@babel/runtime": "*"
```

**Comment by Pessimistress** [comment]
I'm sorry I misread your original issue - yes I agree the package should be listed directly.

Just for context, we are currently working on v9 and considering eliminating `@babel/runtime` by only lightly transpiling the ESM entry point.

**Comment by nemanja-tosic** [comment]
No transpilation for bundlers would probably be the best way going forward in my mind regardless of the import. If v9 is coming soon and there aren't many reports i guess we can wait.

Question out of curiosity and frustration coming from dealing with next.js and SSR recently - when you say ESM entry point, do you think that it would be a good idea to split the entrypoints per consumer, as in using subpath exports node/browser/default, instead of using main/module? node would still need transpilation, probably the minimal supported nodejs version, because SSR (even though the component will not be used on the server) still pulls the code in and would throw if there are unsupported features.

**Comment by Pessimistress** [comment]
The plan is to go full ESM modules with `type: 'module'` with `exports` field in the package.json. The Node/cjs entry point will fix the current SSR issue.

**Comment by thesayyn** [comment]
This is an issue with pnpm as well. the principled fix is to list all of the dependencies, even the ones that are peer.

**Comment by Pessimistress** [comment]
Fixed in v8.9.21.
